packages/puppeteer-core/src/bidi/core/Browser.ts

```diff
@@ -136,15 +134,28 @@ export class Browser extends EventEmitter<{
     return this.#userContexts.values();
   }
 
+  dispose(reason?: string, close?: boolean): void {
+    if (this.disposed) {
+      return;
+    }
+    this.#reason = reason ?? `Browser was disposed.`;
+    if (close) {
+      this.emit('closed', {reason: this.#reason});
+    }
+    this.emit('disconnect', {reason: this.#reason});
+    this.removeAllListeners();
+  }
+
   @throwIfDisposed((browser: Browser) => {
     // SAFETY: By definition of `disposed`, `#reason` is defined.
     return browser.#reason!;
   })
   async close(): Promise<void> {
-    await this.#session.send('browser.close', {});
-    this.#reason = `Browser has already closed.`;
-    this.emit('closed', {reason: this.#reason});
-    this.removeAllListeners();
+    try {
+      await this.#session.send('browser.close', {});
+    } finally {
```

Comment:
Why is this in finally block? The behavior changed now, previously the browser would not be disposed if the command failed.

Reply:
This has not changed, see packages/puppeteer-core/src/bidi/Browser.ts::close